renderer-macos/lib/Styles.re

```diff
@@ -0,0 +1,80 @@
+open CocoaTypes;
+open Layout;
+open BriskStylableText;
+
+type viewStyle = [ | `Border(Border.t) | `Background(Color.t)];
+
+type textStyle = [
+  | `Color(Color.t)
+  | `Font(Font.t)
+  | `Kern(float)
+  | `Align(Alignment.t)
+  | `LineBreak(LineBreak.t)
+  | `LineSpacing(float)
+];
+
+let flushTextStyle = BriskStylableText.applyChanges;
+
+let setViewStyle = (view: view, attr: [> viewStyle]) =>
+  switch (attr) {
+  | `Background(({r, g, b, a}: Color.t)) =>
+    BriskView.setBackgroundColor(view, r, g, b, a)
+  | `Border(({width, radius, color}: Border.t)) =>
+    if (!isUndefined(width)) {
+      BriskView.setBorderWidth(view, width);
+    };
+    if (!isUndefined(radius)) {
+      BriskView.setBorderRadius(view, radius);
+    };
+
+    if (color !== Color.undefined) {
+      let {r, g, b, a}: Color.t = color;
+      BriskView.setBorderColor(view, r, g, b, a);
+    }
+  | _ => ()
+  };
+
+let setTextStyle = (txt: text, attr: [> textStyle]) =>
+  switch (attr) {
+  | `Font(({family, size, weight}: Font.t)) =>
+    let weight =
+      switch (weight) {
+      | `UltraLight => (-0.8)
```

Comment:
These are system-defined constants for `NSFontWeight`.

Reply:
I think this is a bit risky. Now that we have the text abstraction, maybe it'd be good to abstract this out too. This is too error prone.

Reply:
Why risky? I was thinking this will be eventually isolated – you can only pass polymorphic variant.
Check how `Alignment` is done. Note that in `iOS` we'll have a different order as right and left constants are reversed there.

The alternative is passing a variant to Obj-C side that will be converted to int which we'll switch over and still end up with magic `NSFontWeight` aliases.

What's your suggestion?

Reply:
Risky because we depend on some constants that can change at any time.